### ISSUE TITLE
Does not resolve queries through a linking table (Many-To-Many)

### ISSUE THREAD

**Issue by FourSigma** [dosco/super-graph#9]
I have three three tables: `film`, `actor`, and `film_actor` (details below).  The following GraphQL query:  
```
query {
  actor {
    actor_id
    first_name
    last_name
    film {
      film_id
    }
  }
}

# Output 

{
  "error": {
    "error": "something wrong no remote ids found in db response",
    "data": null
  }
}

```
```
                                              Table "public.film"
      Column      |            Type             | Collation | Nullable |                Default                
------------------+-----------------------------+-----------+----------+---------------------------------------
 film_id          | integer                     |           | not null | nextval('film_film_id_seq'::regclass)
 title            | character varying(255)      |           | not null | 
 description      | text                        |           |          | 
 release_year     | year                        |           |          | 
 language_id      | smallint                    |           | not null | 
 rental_duration  | smallint                    |           | not null | 3
 rental_rate      | numeric(4,2)                |           | not null | 4.99
 length           | smallint                    |           |          | 
 replacement_cost | numeric(5,2)                |           | not null | 19.99
 rating           | mpaa_rating                 |           |          | 'G'::mpaa_rating
 last_update      | timestamp without time zone |           | not null | now()
 special_features | text[]                      |           |          | 
 fulltext         | tsvector                    |           | not null | 
Indexes:
    "film_pkey" PRIMARY KEY, btree (film_id)
    "film_fulltext_idx" gist (fulltext)
    "idx_fk_language_id" btree (language_id)
    "idx_title" btree (title)
Foreign-key constraints:
    "film_language_id_fkey" FOREIGN KEY (language_id) REFERENCES language(language_id) ON UPDATE CASCADE ON DELETE RESTRICT
Referenced by:
    TABLE "film_actor" CONSTRAINT "film_actor_film_id_fkey" FOREIGN KEY (film_id) REFERENCES film(film_id) ON UPDATE CASCADE ON DELETE RESTRICT
    TABLE "film_category" CONSTRAINT "film_category_film_id_fkey" FOREIGN KEY (film_id) REFERENCES film(film_id) ON UPDATE CASCADE ON DELETE RESTRICT
    TABLE "inventory" CONSTRAINT "inventory_film_id_fkey" FOREIGN KEY (film_id) REFERENCES film(film_id) ON UPDATE CASCADE ON DELETE RESTRICT
Triggers:
    film_fulltext_trigger BEFORE INSERT OR UPDATE ON film FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('fulltext', 'pg_catalog.english', 'title', 'description')
    last_updated BEFORE UPDATE ON film FOR EACH ROW EXECUTE PROCEDURE last_updated()
```
```
                                            Table "public.actor"
   Column    |            Type             | Collation | Nullable |                 Default                 
-------------+-----------------------------+-----------+----------+-----------------------------------------
 actor_id    | integer                     |           | not null | nextval('actor_actor_id_seq'::regclass)
 first_name  | character varying(45)       |           | not null | 
 last_name   | character varying(45)       |           | not null | 
 last_update | timestamp without time zone |           | not null | now()
Indexes:
    "actor_pkey" PRIMARY KEY, btree (actor_id)
    "idx_actor_last_name" btree (last_name)
Referenced by:
    TABLE "film_actor" CONSTRAINT "film_actor_actor_id_fkey" FOREIGN KEY (actor_id) REFERENCES actor(actor_id) ON UPDATE CASCADE ON DELETE RESTRICT
Triggers:
    last_updated BEFORE UPDATE ON actor FOR EACH ROW EXECUTE PROCEDURE last_updated()
```
```
                         Table "public.film_actor"
   Column    |            Type             | Collation | Nullable | Default 
-------------+-----------------------------+-----------+----------+---------
 actor_id    | smallint                    |           | not null | 
 film_id     | smallint                    |           | not null | 
 last_update | timestamp without time zone |           | not null | now()
Indexes:
    "film_actor_pkey" PRIMARY KEY, btree (actor_id, film_id)
    "idx_fk_film_id" btree (film_id)
Foreign-key constraints:
    "film_actor_actor_id_fkey" FOREIGN KEY (actor_id) REFERENCES actor(actor_id) ON UPDATE CASCADE ON DELETE RESTRICT
    "film_actor_film_id_fkey" FOREIGN KEY (film_id) REFERENCES film(film_id) ON UPDATE CASCADE ON DELETE RESTRICT
Triggers:
    last_updated BEFORE UPDATE ON film_actor FOR EACH ROW EXECUTE PROCEDURE last_updated()
```

**Comment by dosco** [comment]
I'm almost certain you are running an older version of the code. I recall fixing an issue related to this error a couple days ago. Can you try `GO111MODULE=on go get -u github.com/dosco/super-graph`

**Comment by FourSigma** [comment]
Still having issues with this one despite updating to the latest version of super-graph

**Comment by dosco** [comment]
Is this a database I can download somewhere. `go get` is weird sometimes it does not update I might need to change how I tag versions. If I can get that db I can give it a try but I'm almost certain you're not on the latest version. 

I need to figure out a way to add a version to the binaries and still have `go get` build the binary instead of a build script

**Comment by FourSigma** [comment]
Hmm...possibly. Putting versions into binaries should be  fairly simple.  I have seen Go binaries with the Git hashes in them and versions done on build.  Maybe I will put in a PR for this tomorrow or next week. 

Here is the database: https://github.com/robconery/dvdrental 

It could be something  up with the database schema.  Easy script to restore. I was going to use this database in the `examples` folder.

**Comment by dosco** [comment]
Did you get a chance to try this with the latest version. Looking to close this issue as I think it's fixed.

**Comment by FourSigma** [comment]
Vikram, I haven’t had a chance to check it out yet.  I will try and get to
it this weekend.

**Comment by dosco** [comment]
The latest version has several bug fixes that might have fixed this issue.  Also if you don't use foreign keys within your db (Super Graph uses these to discover relationships) then you can now add them through the config file. https://supergraph.dev/guide.html#database-relationships

```
Super Graph v0.12.5-11-g2d3e3cb
For documentation, visit https://supergraph.dev

Commit SHA-1          : 2d3e3cb
Commit timestamp      : 2019-12-09 01:59:30 -0500
Branch                : master
Go version            : go1.13
```

**Comment by Pestdoktor** [comment]
I also encountered this and did some debugging.

Using the following patch I can query the *linking* table:
```diff --git a/psql/tables.go b/psql/tables.go
index 5c78d98..619962e 100644
--- a/psql/tables.go
+++ b/psql/tables.go
@@ -194,6 +194,17 @@ ORDER BY id;`
 			if c.Array {
 				v.Array = true
 			}
+			if c.FKeyTable != "" {
+				v.FKeyTable = c.FKeyTable
+			}
+			if c.fKeyColID.Elements != nil {
+				v.fKeyColID = c.fKeyColID
+				err := v.fKeyColID.AssignTo(&v.FKeyColID)
+				if err != nil {
+					return nil, err
+				}
+			}
+			cmap[c.ID] = v
 		} else {
 			err := c.fKeyColID.AssignTo(&c.FKeyColID)
 			if err != nil {
```
**Query:**
```query {
  users{
    id
    many_user_has_many_permissions{id_user}
  }
}
```
**Result:**
```{
  "data": {
    "users": [
      {
        "id": 1,
        "many_user_has_many_permissions": [
          {
            "id_user": 1
          }
        ]
      }
    ]
  }
}
```
---
But I get another error when trying to query the *linked* table:
**Query:**
```query {
  users{
    id
    permissions{id}
  }
}
```
**Result:**
```{
  "error": {
    "message": "ERROR: column user.id_user does not exist (SQLSTATE 42703)"
  }
}
```
I'll probably be able to investigate further tomorrow

**Comment by dosco** [comment]
@Pestdoktor thats a great catch I think I introduced that bug when refactored from using `map[int16]*DBColumn` to `map[int16]DBColumn`. I have added your patch and committed a fix. I'm not sure how to give you credit for the fix other than on here. 

I noticed your query above uses `id_user` whats that is it a primary key column? Also what is your table name is it `many_user_has_many_permissions` or `permissions`? Can you share your schema here.

```
query {
  users{
    id
    many_user_has_many_permissions{id_user}
  }
}
```

**Comment by dosco** [comment]
For me right now using the built-in example app I get the following result

**Launch example app**
```
docker-compose run rails_app rake db:create db:migrate db:seed
docker-compose up
open http://localhost:8080
```

**GraphQL Query**
```
query {
  users {
    id
    email
    picture: avatar
    products(limit: 2, where: {price: {gt: 10}}) {
      id
      name
      description
    }
  }
}
```

**Response**
```
{
  "data": {
    "users": [
      {
        "id": 1,
        "email": "user0@demo.com",
        "picture": "https://robohash.org/quibeataedeleniti.png?size=300x300&set=set1",
        "products": [
          {
            "id": 7,
            "name": "St. Bernardus Abt 12",
            "description": "Heineken, Bock, Ahtanum, 1332 - Northwest Ale, 32 IBU, 4.8%, 10.9°Blg"
          },
          {
            "id": 24,
            "name": "Oaked Arrogant Bastard Ale",
            "description": "Red Stripe, Light Hybrid Beer, Hallertau, 1332 - Northwest Ale, 22 IBU, 4.8%, 9.4°Blg"
          }
        ]
      },
      {
        "id": 2,
        "email": "user1@demo.com",
        "picture": "https://robohash.org/iureconsequatursed.png?size=300x300&set=set1",
        "products": [
          {
            "id": 15,
            "name": "Sierra Nevada Bigfoot Barleywine Style Ale",
            "description": "Samuel Adams, Light Lager, Millennium, 1028 - London Ale, 70 IBU, 6.0%, 18.3°Blg"
          },
          {
            "id": 16,
            "name": "Péché Mortel",
            "description": "Kirin, Amber Hybrid Beer, Northern Brewer, 2206 - Bavarian Lager, 57 IBU, 8.1%, 16.3°Blg"
          }
        ]
      },
...
```

**Comment by Pestdoktor** [comment]
The schema I'm using is two tables linked by many-to-many as output by [pgModeler](https://pgmodeler.io/):
```
-- Database generated with pgModeler (PostgreSQL Database Modeler).
-- pgModeler  version: 0.9.2-beta2
-- PostgreSQL version: 12.0
-- Project Site: pgmodeler.io
-- Model Author: ---


-- Database creation must be done outside a multicommand file.
-- These commands were put in this file only as a convenience.
-- -- object: aktionauto | type: DATABASE --
-- -- DROP DATABASE IF EXISTS aktionauto;
-- CREATE DATABASE aktionauto;
-- -- ddl-end --
-- 

-- object: public."user" | type: TABLE --
-- DROP TABLE IF EXISTS public."user" CASCADE;
CREATE TABLE public."user" (
	id smallint NOT NULL GENERATED ALWAYS AS IDENTITY ,
	mail character varying(60),
	password character varying(60),
	CONSTRAINT user_mail_unique UNIQUE (mail),
	CONSTRAINT user_pk PRIMARY KEY (id)

);
-- ddl-end --
ALTER TABLE public."user" OWNER TO postgres;
-- ddl-end --

-- object: public.permission | type: TABLE --
-- DROP TABLE IF EXISTS public.permission CASCADE;
CREATE TABLE public.permission (
	id smallint NOT NULL GENERATED ALWAYS AS IDENTITY ,
	permission character varying(32),
	CONSTRAINT permission_permission_unique UNIQUE (permission),
	CONSTRAINT permission_pk PRIMARY KEY (id)

);
-- ddl-end --
ALTER TABLE public.permission OWNER TO postgres;
-- ddl-end --

-- object: public.many_user_has_many_permission | type: TABLE --
-- DROP TABLE IF EXISTS public.many_user_has_many_permission CASCADE;
CREATE TABLE public.many_user_has_many_permission (
	id_user smallint NOT NULL,
	id_permission smallint NOT NULL,
	CONSTRAINT many_user_has_many_permission_pk PRIMARY KEY (id_user,id_permission)

);
-- ddl-end --

-- object: user_fk | type: CONSTRAINT --
-- ALTER TABLE public.many_user_has_many_permission DROP CONSTRAINT IF EXISTS user_fk CASCADE;
ALTER TABLE public.many_user_has_many_permission ADD CONSTRAINT user_fk FOREIGN KEY (id_user)
REFERENCES public."user" (id) MATCH FULL
ON DELETE RESTRICT ON UPDATE CASCADE;
-- ddl-end --

-- object: permission_fk | type: CONSTRAINT --
-- ALTER TABLE public.many_user_has_many_permission DROP CONSTRAINT IF EXISTS permission_fk CASCADE;
ALTER TABLE public.many_user_has_many_permission ADD CONSTRAINT permission_fk FOREIGN KEY (id_permission)
REFERENCES public.permission (id) MATCH FULL
ON DELETE RESTRICT ON UPDATE CASCADE;
-- ddl-end --

-- object: pgcrypto | type: EXTENSION --
-- DROP EXTENSION IF EXISTS pgcrypto CASCADE;
CREATE EXTENSION pgcrypto
WITH SCHEMA public;
-- ddl-end --
```

**Comment by Pestdoktor** [comment]
I found a fix in #27

**Comment by dosco** [comment]
This is awesome. I tested it works great.
```
query {
  user {
    id
    mail
    permission {
      id
      permission
    }
  }
}
```

```
{
  "data": {
    "user": {
      "id": 1,
      "mail": "test@test.co",
      "permission": {
        "id": 1,
        "permission": "read_write"
      }
    }
  }
}
```
